Clean up toolbox

ReneSkukies opened the pull request:
<!--
Thanks for making a pull request to UnfoldRIDE.jl.
We have added this PR template to help you help us.
Make sure to read the contributing guidelines and abide by the code of conduct.
See the comments below, fill the required fields, and check the items.
-->

## Related issues

<!-- We normally work with (i) create issue; (ii) discussion if necessary; (iii) create PR. So, at least one of the following should be true:-->

<!-- Option 1, this closes an existing issue. Fill the number below-->
Closes #36  #37 

<!-- Option 2, this is a small fix that arguably won't need an issue. Uncomment below -->
<!--
There is no related issue.
-->

## Checklist

<!-- mark true if NA -->
<!-- leave PR as draft until all is checked -->
- [x ] I am following the [contributing guidelines](https://github.com/unfoldtoolbox/UnfoldRIDE.jl/blob/main/docs/src/90-contributing.md)
- [ x] Tests are passing
- [ ] Lint workflow is passing
- [ x] Docs were updated and workflow is passing

ReneSkukies committed:
beginning of clean up

ReneSkukies committed:
add functionality to provide own formula for UnfoldRIDE

ReneSkukies committed:
fix typos, add formula to docs and readme

ReneSkukies committed:
add compats

behinger commented on src/ride/ride_structs.jl:
```diff
@@ -39,6 +41,7 @@ cfg = RideConfig(
     r_range = [0, 0.8],
     c_range = [-0.4, 0.4],
     c_estimation_range = [-0.1, 0.9],
+    formulas = [@formula(0 ~ 1), @formula(0 ~ 1), @formula(0 ~ 1 + reaction_time)] # formulas used for S, R, and C component
```
would it make sense to do:
`formulas = ['S'=>@formula(0~1),'C'=>@formula(0~1),'R'=>@formula(0~1)`], e.g. as a list of pairs? that would generalize to more events more readily and doesnt have the issue of order. You the  only have to splice in the firbasis in the fit command down below

behinger reviewed (commented):
Some further comments:
- Any reason to have the ride methods in a folder ride?
- Any reason that the ride methods have a `ride_` in the filenames?
- The docs currently dont show any plots
- you seem to have still quite some code doubling in `ride_classic_algorithm` and `ride_unfold_algorithm`. I mostly checked the "prepare_data" phase, which reads near 1:1. I generally would prefer to have one "master"-loop?+´
- I think similarities between both approaches would appear starkly, once the code is refactored into smaller functions. Right now it is a very large near-monolithic function

I didnt look much more into the functions, because if you decide to put both methods together, I wouldnt need to review two separate files :)

ReneSkukies committed:
Update Project.toml

ReneSkukies committed:
Introduce rounding when saving latencies

ReneSkukies committed:
Whoops...

ReneSkukies committed:
Add model to return values in ride_unfold_methods

ReneSkukies committed:
Modify return values to include final model

Updated the return values of the function to include the final model.

ReneSkukies committed:
Fix variable name for final model return

ReneSkukies committed:
Delete for now until I have time to fix this properly

ReneSkukies committed:
Fix bug in test due to unfoldsim bug

ReneSkukies committed:
Return final UnfoldModel

Now returns the final model of the Unfold algorithm, also fixes tests

ReneSkukies committed:
Fix range calculation in ride_classic_methods.jl

ReneSkukies committed:
Fix dsp bug/ code doubling/ docstrings

This fixes the dsp sfreq bug and gets rid of quite some code doubling between the methods.

Sorry for the gigantic commit....

ReneSkukies committed:
Merge pull request #38 from unfoldtoolbox/clean_up_agent_test

Fix dsp bug/ code doubling/ docstrings

ReneSkukies committed:
Update Julia version compatibility in Project.toml
